Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedCredential.cs

```diff
@@ -34,11 +34,20 @@ namespace Google.Apis.Auth.OAuth2
     /// and https://cloud.google.com/iam/docs/impersonating-service-accounts
     /// for more information.
     /// </summary>
-    public sealed class ImpersonatedCredential : ServiceCredential, IOidcTokenProvider, IGoogleCredential, IBlobSigner
+    public sealed class ImpersonatedCredential : BasicImpersonatedCredential, IGoogleCredential, IOidcTokenProvider, IBlobSigner
```

Comment:
Ah, so this is why BasicImpersonatedCredential needs to be public.
Could it be abstract, and renamed to ImpersonatedCredentialBase?